kgraphql/src/main/kotlin/com/apurebase/kgraphql/GraphQLError.kt

```diff
@@ -39,26 +44,20 @@ open class GraphQLError(
      * The original error thrown from a field resolver during execution.
      */
     val originalError: Throwable? = null,
-    val extensionsErrorType: String? = "INTERNAL_SERVER_ERROR",
-    val extensionsErrorDetail: Map<String, Any?>? = null
-) : Exception(message) {
 
-    constructor(message: String, node: ASTNode?) : this(message, nodes = node?.let(::listOf))
-
-    constructor(message: String, extensionsErrorType: String?) : this(
-        message,
-        null,
-        null,
-        null,
-        null,
-        extensionsErrorType
-    )
+    /**
+     * The type of the error, based on Apollo Server's built-in error codes:
+     *   https://www.apollographql.com/docs/apollo-server/data/errors#built-in-error-codes
+     *
+     * For supported built-in codes see [BuiltInErrorCodes].
+     */
+    val extensionsErrorType: String,
 
-    constructor(
-        message: String,
-        extensionsErrorType: String?,
-        extensionsErrorDetail: Map<String, Any?>?
-    ) : this(message, null, null, null, null, extensionsErrorType, extensionsErrorDetail)
+    /**
+     * Details regarding this error.
+     */
+    val extensionsErrorDetail: Map<String, Any?>? = null
+) : Exception(message) {
```

Comment:
Shouldn't this rather be `RuntimeException`?

Reply:
Not sure. What would be the relevant difference? https://kotlinlang.org/docs/exceptions.html#exception-hierarchy gives me mixed feelings. 

On the one hand

> `RuntimeException` is usually caused by insufficient checks in the program code and can be prevented programmatically.

but on the other hand, `IllegalArgumentException` is also a `RuntimeException`.

Reply:
Looking at this (https://kotlinlang.org/docs/exceptions.html#exception-hierarchy) I'd agree with `Exception.`

But this
> but on the other hand, IllegalArgumentException is also a RuntimeException

confuses me as well.

I'd say from the definition in the link it depends
* `ExecutionException` could very well be a sub-class of `RuntimeException`, because this most likely is caused by a bug in either the library or the schema definition
* The rest could very well be a sub-class of `Exception`, since I think all of them depend on the received query (?)

Reply:
But reading what you stated it can most likely stay as it is as well, so I'll accept it.

I'm thinking whether I personally would want a variant of `Schema::execute` to return me something like either an Error or the successful result.

I'm more a fan of typed error handling instead of exceptions.

Maybe it would be good to provide both, since there might very well be people who think differently and prefer exceptions.

But that is outside of the scope of this PR.